kernel/unwind/src/lsda.rs

```diff
@@ -284,11 +284,11 @@ impl<R: Reader> FallibleIterator for CallSiteTableIterator<R> {
 fn read_encoded_pointer<R: gimli::Reader>(reader: &mut R, encoding: DwEhPe) -> gimli::Result<u64> {
     match encoding {
         DW_EH_PE_omit     => Err(gimli::Error::CannotParseOmitPointerEncoding),
-        DW_EH_PE_absptr   => reader.read_u64().map(|v| v as u64),
-        DW_EH_PE_uleb128  => reader.read_uleb128().map(|v| v as u64),
+        DW_EH_PE_absptr   => reader.read_u64().map(|v| v ), // v:u64
```

Comment:
all of these `.map(|v| v)` are what's causing the current clippy CI pass to fail, since they are now useless maps. So you can remove them.